### DISCUSSION TITLE
Testing instructions are outdated for newer Android Studio versions

### DISCUSSION THREAD

**Discussion by Bontrey** [oppia/oppia-android#5184] [Q&A (Other)] [answered]
macOS 13.6 Ventura

Android Studio Giraffe 2022.3.1 Patch 1

According to the [testing instructions](https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing/35c374468bb6a6c1d8e1a98988b3d6eddaf2239e#running-robolectric-tests) 'Android Unit' should be used, but that no longer exists, and selecting just 'JUnit' and then any module, such as 'oppia-android.app.unitTest' won't be able to find any of the `app/` tests (at runtime though; it validates in the configuration as being part of the module). I tried it with StateFragmentTest

What does seem to work is a [gradle test](https://github.com/oppia/oppia-android/assets/698755/6c8e9a10-e70e-457e-adf8-a98fa413aed0). I'm not sure where symbols like `testDebugUnitTest` come from, but at least the tests run.

And running Android Instrumented Tests, which I'm assuming are the Espresso tests, runs 0 tests if my configuration is [like this](https://github.com/oppia/oppia-android/assets/698755/89108b69-fc20-4126-8c3e-925b61693e2e). Selecting a specific class does not help, tried it with StateFragmentTest again (some of the tests are annotated as Espresso-only, so assuming that they are Espresso tests). Playing around with the configurations generated by right-click options on the arrows next to the line numbers in test files also turned up with nothing that ran an Espresso test.

Would be great if someone with more experience with the project explained the proper testing setup for a newish Android Studio version. Thanks

EDIT: M1 mac. tried both API 29 (installation guide said to use this for ARM) and 28 emulators

**Comment by adhiamboperes** [comment, marked as answer]
Thanks @Bontrey for filing this detailed report. I have not run robolectric tests on Giraffe yet, but you probably will not be able to run espresso tests because of our test configuration and gradle compatibility reasons. Could you please take a look at our updated set up instructions for an alternative development environment? https://github.com/oppia/oppia-android/wiki/Installing-Oppia-Android

**Reply by Bontrey** [reply]
Using bumblebee works. Although some tests seem to time out due to one animation or another. I'll keep tinkering with it. Thanks
